vignettes/rhino-fluent.Rmd

```diff
@@ -0,0 +1,560 @@
+
+```{r, echo=FALSE}
+knitr::include_graphics("images/rhino_fluent_3.png")
+```
+
+## Add a simple toggle input feature for the table with basic styling
+
+Till now, we could only view table with all the data it has.
+We will now add a functionality to view those rows in the table where value of the "Is Closed?" column is 1 by default and provide a toggle to the user to also include results where the value is 0.
+The "Is Closed?" column basically indicates whether the deal is closed or not with values 0 and 1 where 0 means False/No and 1 means True/Yes.
+
+We will make use of the `dplyr` package now. We will use [`Toggle.shinyInput`](https://appsilon.github.io/shiny.fluent/reference/Toggle.html) from `shiny.fluent` package which represents a physical switch that allows someone to choose between two mutually exclusive options. For example,"On/Off","Show/Hide". Choosing an option should produce an immediate result.
+
+In the UI, we include the code for showing the toggle for input - `Toggle.shinyInput(ns("includeOpen"), label = "Include open deals"),` and work on the input provided inside the server section
+
+So, the `app/view/datatable.R` file now is modified to -
+
+```         
+# app/view/datatable.R
+
+box::use(
+  dplyr[filter],
+  shiny[moduleServer, NS, div, uiOutput, renderUI, reactive],
+  shiny.fluent[fluentPage, Text, DetailsList, Toggle.shinyInput],
+  tibble[tibble]
+)
+
+columns <- tibble(
+  fieldName = c("rep_name", "date", "deal_amount", "city", "is_closed"),
+  name = c("Sales Rep", "Date", "Deal Amount", "City", "Is Closed?")
+)
+
+#' @export
+ui <- function(id) {
+  ns <- NS(id)
+  div(
+    Text("A randomly generated dataset of imaginary sales deals", variant = "large"),
+    div(
+      class = "ms-depth-4",
+      Toggle.shinyInput(ns("includeOpen"), label = "Include open deals")
+    ),
+    uiOutput(ns("datatable")),
+  )
+}
+
+#' @export
+server <- function(id) {
+  moduleServer(id, function(input, output, session) {
+    filtered_deals <- reactive({
+      shiny.fluent::fluentSalesDeals |> filter(
+        is_closed | input$includeOpen
+      )
+    })
+    output$datatable <- renderUI({
+      DetailsList(items = filtered_deals(), columns = columns)
+    })
+  })
+}
+```
+
+This code is defining a reactive expression called `filtered_deals` in an R Shiny application.
+
+The reactive expression uses the `|>` operator to pipe the `shiny.fluent::fluentSalesDeals` data frame to the `filter()` function which comes from the `dplyr` package.
+The `filter()` function is used to filter rows from a data frame based on one or more conditions.
+
+In this case, the `filter()` function is filtering rows from `shiny.fluent::fluentSalesDeals` based on two conditions:
+
+-   `is_closed`: This is a logical column in the `shiny.fluent::fluentSalesDeals` data frame that indicates whether a sales deal has been closed or not.
```

Comment:
This part is too repetitive, I don't see a point in elaborating that function named `filter` does filtering.